lib/elixir/lib/node.ex

```diff
@@ -92,12 +93,49 @@ defmodule Node do
 
   Inlined by the compiler.
   """
-  @type state :: :visible | :hidden | :connected | :this | :known
   @spec list(state | [state]) :: [t]
   def list(args) do
     :erlang.nodes(args)
   end
 
+  @doc """
+  Returns a list of nodes according to argument given, with extra info.
+
+  The second argument is used to request additional information with the
+  following options:
+
+  * `:connection_id` - boolean. Returns the connection id from each node.
+  * `:node_type` - boolean. Returns the node type (or state) of each node.
+
+  The additional information is returned as a keyword.
+
+  ## Examples
+
+      iex> Node.list([:connected, :this], [connection_id: true])
```

Comment:
It's probably better to use the syntax that the formatter would give?
```suggestion
      iex> Node.list([:connected, :this], connection_id: true)
```